cairo-verifier/src/utils.rs

```diff
@@ -0,0 +1,223 @@
+use cairo_platinum_prover::{air::{generate_cairo_proof, verify_cairo_proof, PublicInputs}, cairo_mem::CairoMemory, execution_trace::build_main_trace, register_states::RegisterStates};
+use hyle_contract::HyleOutput;
+use stark_platinum_prover::proof::options::{ProofOptions, SecurityLevel};
+use serde::{Deserialize, Serialize};
+use lambdaworks_math::field::fields::fft_friendly::stark_252_prime_field::Stark252PrimeField;
+use stark_platinum_prover::proof::stark::StarkProof;
+use error::VerifierError;
+use num::BigInt;
+
+pub mod error;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Event {
+    from: String,
+    to: String,
+    amount: u64,
+}
+
+pub fn verify_proof(proof_path: &String) -> Result<String, VerifierError>{
+    let proof_options = ProofOptions::new_secure(SecurityLevel::Conjecturable100Bits, 3);
+    let Ok(program_content) = std::fs::read(proof_path) else {
+        return Err(VerifierError(format!("Error opening {} file", proof_path)));
+    };
+    let mut bytes = program_content.as_slice();
+    if bytes.len() < 8 {
+        return Err(VerifierError(format!("Error reading proof from file: {}", proof_path)));
+    }
+
+    // Proof len was stored as an u32, 4u8 needs to be read
+    let proof_len = u32::from_le_bytes(bytes[0..4].try_into().unwrap()) as usize;
+
+    bytes = &bytes[4..];
+    if bytes.len() < proof_len {
+        return Err(VerifierError(format!("Error reading proof from file: {}", proof_path)));
+    }
+    
+    let Ok((proof, _)) = bincode::serde::decode_from_slice(
+        &bytes[0..proof_len],
+        bincode::config::standard(),
+    ) else {
+        return Err(VerifierError(format!("Error reading proof from file: {}", proof_path)));
+    };
+    
+    // PublicInputs len was stored as an u32, 4u8 needs to be read
+    let pub_inputs_len = u32::from_le_bytes(bytes[proof_len..proof_len+4].try_into().unwrap()) as usize;
+    let pub_inputs_bytes = &bytes[proof_len+4..proof_len+4+pub_inputs_len];
+    
+    let Ok((pub_inputs, _)) =
+    bincode::serde::decode_from_slice(pub_inputs_bytes, bincode::config::standard())
+    else {
+        return Err(VerifierError(format!("Error reading proof from file: {}", proof_path)));
+    };
+    let program_output_bytes = &bytes[proof_len+4+pub_inputs_len..];
+
+    let Ok((program_output, _)) =
+        bincode::serde::decode_from_slice::<HyleOutput<Event>, _>(program_output_bytes, bincode::config::standard())
+    else {
+        return Err(VerifierError(format!("Error reading proof from file: {}", proof_path)));
+    };
+
+    if verify_cairo_proof(&proof, &pub_inputs, &proof_options){
```

Comment:
We extract:

1. the inputs
2. the program output
3. the proof
from the file, each encoded with the length l (encoded as u32) needed to know on how many bytes the field is stored.
The output seems independent from the 2 first fields as it is not used during verification ?

Reply:
As we talked, for the moment  we do not extract output from the proof